.github/workflows/build.yaml

```diff
@@ -0,0 +1,165 @@
+        working-directory: ${{env.ProtonBuildDir}}
+        run: cmake "${{github.workspace}}/qpid-proton" "-DCMAKE_INSTALL_PREFIX=${InstallPrefix}" "-DCMAKE_BUILD_TYPE=${BuildType}" "-DBUILD_TESTING=OFF" "-DENABLE_FUZZ_TESTING=OFF" "-GNinja" ${ProtonCMakeExtraArgs}
+        shell: bash
+      - name: qpid-proton cmake build/install
+        run: cmake --build "${ProtonBuildDir}" --config ${BuildType} -t install
+        shell: bash
+      - name: qpid-dispatch cmake configure
+        working-directory: ${{env.DispatchBuildDir}}
+        run: cmake "${{github.workspace}}/qpid-dispatch" "-DCMAKE_INSTALL_PREFIX=${InstallPrefix}" "-DCMAKE_BUILD_TYPE=${BuildType}" "-GNinja" "-DUSE_BWRAP=ON" ${DispatchCMakeExtraArgs}
+        shell: bash
+      - name: qpid-dispatch cmake build/install
+        run: cmake --build "${DispatchBuildDir}" --config ${BuildType} -t install
+        shell: bash
+      # github actions/upload-artifact@v2 does not preserve file permissions (executable)
+      - name: Compress build
+        working-directory: ${{github.workspace}}
+        run: tar --xz -cf /tmp/archive.tar.xz --exclude '*.o' --exclude '*.pyc' --exclude '.git' --exclude='qpid-dispatch/build/console' qpid-dispatch install qpid-proton/build/python/pkgs
+      - name: Upload archive
+        uses: actions/upload-artifact@v2
+        with:
+          name: qpid_dispatch_wrk_${{matrix.os}}_${{matrix.buildType}}
+          path: /tmp/archive.tar.xz
+  test:
+    name: "Test (${{matrix.os}}, shard ${{matrix.shard}} of ${{matrix.shards}}"
+    runs-on: ${{ matrix.os }}
+    needs: [build]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04]
+        buildType: [RelWithDebInfo]
+        shard: [1, 2, 3, 4]
+        shards: [4]
+    env:
+      BuildType: ${{matrix.buildType}}
+      ProtonBuildDir: ${{github.workspace}}/qpid-proton/build
+      DispatchBuildDir: ${{github.workspace}}/qpid-dispatch/build
+      InstallPrefix: ${{github.workspace}}/install
+      PKG_CONFIG_PATH: ${{matrix.pkg_config_path}}
+      LD_LIBRARY_PATH: ${{github.workspace}}/install/lib
+    steps:
+      - name: Download Build
+        uses: actions/download-artifact@v2
+        with:
+          name: qpid_dispatch_wrk_${{matrix.os}}_${{matrix.buildType}}
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+          architecture: x64
+      - name: Install python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox
+      - name: Install Linux dependencies
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          sudo apt install -y bubblewrap swig ninja-build libsasl2-2 sasl2-bin libjsoncpp1 libwebsockets15 xz-utils
+      - name: Install MacOS dependencies
+        if: ${{ runner.os == 'macOS' }}
+        run: |
+          brew install cmake ninja libuv swig pkgconfig jsoncpp openssl libwebsockets
```

Comment:
A little forward-looking perhaps given the job doesnt run there yet? :)

Reply:
I'll drop it. I started doing both Linux and macOS, and then reduced the scope.